llvm/lib/Target/AMDGPU/AMDGPUCodeGenPrepare.cpp

```diff
@@ -369,9 +450,269 @@ bool AMDGPUCodeGenPrepareImpl::run(Function &F) {
       }
     }
   }
+
+  // GlobalISel should directly use the values, and do not need to emit
+  // CopyTo/CopyFrom Regs across blocks
+  if (UsesGlobalISel)
```

Comment:
This one isn't easy because of how the fallback works. The full pipeline always runs, the IR optimizations aren't redone after the fallback hits

Reply:
I thought this was just a consequence of local ISel -- local ISel needs to use CopyTo/FromRegs to propagate values across blocks, and, in this process, legalizes the values. 

However, it appears that GlobalISel is doing something similar.